main.cc

```diff
@@ -433,6 +433,39 @@ static int scylla_main(int ac, char** av) {
         exit(1);
     }
 
+    // Even on the environment which causes error during initalize Scylla,
+    // "scylla --version" should be able to run without error.
+    // To do so, we need to parse and execute these options before
+    // initializing Scylla/Seastar classes.
+    bpo::options_description preinit_description("Scylla options");
+    bpo::variables_map preinit_vm;
+    preinit_description.add_options()
+        ("version", bpo::bool_switch(), "print version number and exit")
+        ("build-id", bpo::bool_switch(), "print build-id and exit")
+        ("build-mode", bpo::bool_switch(), "print build mode and exit")
+        ("list-tools", bpo::bool_switch(), "list included tools and exit");
+    auto preinit_parsed_opts = bpo::command_line_parser(ac, av).options(preinit_description).allow_unregistered().run();
+    bpo::store(preinit_parsed_opts, preinit_vm);
+    if (preinit_vm["version"].as<bool>()) {
+        fmt::print("{}\n", scylla_version());
+        return 0;
+    }
+    if (preinit_vm["build-id"].as<bool>()) {
+        fmt::print("{}\n", get_build_id());
+        return 0;
+    }
+    if (preinit_vm["build-mode"].as<bool>()) {
+        fmt::print("{}\n", scylla_build_mode());
+        return 0;
+    }
+    if (preinit_vm["list-tools"].as<bool>()) {
+        fmt::print(
+                "types - a command-line tool to examine values belonging to scylla types\n"
+                "sstable - a multifunctional command-line tool to examine the content of sstables\n"
+        );
+        return 0;
+    }
```

Comment:
Maybe we should move these to `main()` (the caller of `scylla_main`)?

Reply:
Seems like `scylla_types_main()` and `scylla_sstable_main()` have different options and does not have `--version` option, so I think head of `scylla_main()` is correct place.

see:
https://github.com/scylladb/scylladb/blob/master/tools/scylla-types.cc#L185
https://github.com/scylladb/scylladb/blob/master/tools/scylla-sstable.cc#L2025

Reply:
They don't have separate `--version` because they are the same executable as `scylla` and therefore have the same version. A common --version does make sense.

Reply:
Yes. We can do that as a follow-up.

Reply:
> They don't have separate `--version` because they are the same executable as `scylla` and therefore have the same version. A common --version does make sense.

Okay, I will move it to main.